### ISSUE TITLE
Discrepancy for between minimap2 and mm2-fast for assembly alignment

### ISSUE THREAD

**Issue by jdwinkler** [bwa-mem2/mm2-fast#5]
Hi,

Thanks for your work on this optimized version of minimap2, I think it is quite useful. I was recently building out a pipeline involving the pairwise alignment of possibly divergent (~75% ANI) assemblies and noticed that minimap2 and mm2-fast do not give the same result. The underlying command was:

minimap2  -x asm20 assembly_a assembly_b

I know this is not really the intended purpose of minimap2, but I thought it strange that the results were not consistent between your fork and the main branch. I can find example NCBI assemblies to give different results if you want to look into it?

**Comment by jdwinkler** [comment]
I will also check to make sure I am compare like versions as well, that could potentially be the source of the problem I think.

edit: at 2.24 for both tools, I see different alignments when comparing fragmented MAGs.

**Comment by joydddd** [comment]
Hi jdwinkler, we're working on accelerating pairwise alignment on GPU and I'm also running into inconsistent results between minimap2 2.24, the algorithm described in the Minimap2 paper, and this mm2-fast kernel. I'm digging into these divergent results. Could you share some example NCBI assemblies with me?

**Comment by jdwinkler** [comment]
Yes, let me dig those up tomorrow US time and I'll send you a couple public assemblies.

**Comment by sagnikbanerjee15** [comment]
Hello,

I am using this command to generate alignments.

minimap2 -H -a -o XXX.sam -x splice -t 64 -Y --eqx --MD --cs=long --max-chain-skip=1000000 genome.fasta XXX.fastq

They give different number of alignments for the normal minimap2 vs the fast version. Could you please look into it?

Thanks
